mixer/cmd/bundles.go

```diff
@@ -114,10 +114,55 @@ var bundleListCmd = &cobra.Command{
 	},
 }
 
+// Bundle Edit command ('mixer bundle edit')
+type bundleEditCmdFlags struct {
+	noEdit bool
+	editor string
+	add    bool
+}
+
+var bundleEditFlags bundleEditCmdFlags
+
+var bundleEditCmd = &cobra.Command{
+	Use:   "edit [bundle(s)]",
+	Short: "Edit local and upstream bundles",
+	Long: `Edit local and upstream bundle definition files. This command will locate the
+bundle (looking first in local-bundles, then in upstream-bundles), and launch
+an editor to edit it. If the bundle is only found upstream, the bundle file will
+first be copied to your local-bundles directory for editing. When the editor
+closes, the bundle file is then parsed for validity.
+
+The editor is configured in builder.conf via the BUNDLE_EDITOR field, or by flag
+override. Valid options include: nano, vim, emacs, joe, and ctags. The default
+value is nano.
+
+Passing '--no-edit' will suppress launching the editor, and will thus only copy
+the bundle file to local-bundles if it is only found upstream. This can be
+useful if you want to add a bundle to local-bundles, but wish to edit it at a
```

Comment:
Is this a real use case? Would it be better as a different command?

Once you allow someone to choose their own editor, they can always use /bin/true to get the same effect.

Reply:
Honestly, this is something people asked for. I don't remember exactly who or when, but when I proposed this feature back in December, this is the feedback I got.

Also, while yes -- if you could pick any editor you want, you could use `/bin/true` -- expecting users to know or do that is the opposite of user-friendly.

Reply:
I think that is an important part of the feature, because that is a boring computer task that mixer can do for us: hey just get me the desktop and desktop-apps from the upstream. I've almost suggested something like `mixer bundle copy desktop desktop-apps` (or copy-upstream, or import, or whatever) as a standalone thing (so you don't have to deal with the "edit but no edit").

Reply:
@cmarcelo I had considered at one point making that a stand-alone `mixer bundle import` command, or even a flag off `add` (`mixer bundle add --local <bundle>`).

The reason I don't want that is that, while people might very well find it useful to pull things into `local-bundles` and edit at some later time, I want them to still be aware that doing this _overrides_ the upstream version. If you're not actually planning to edit the bundle, having a copy in `local-bundles` is almost certainly just going to cause confusion later (either for you, or for the tool when it tries to migrate to new upstream version). So keeping it a part of the `mixer bundle edit` command reminds people why they're moving things into `local-bundles`: to edit them.

On that note, one (potentially valid, but dangerous) use-case for copying into local and _not_ editing is exactly that -- persisting a bundle past a deletion in a future upstream version. What I mean is, if you add bundle_a to `local-bundles`, and in a future version of upstream bundle_a gets removed entirely, it now becomes equivalent to a "user-created" bundle -- it is no longer masking anything upstream. That very well might be a desirable thing for someone, but for most people that will likely just cause confusion.